bin/db_backup_initial.rb

```diff
@@ -1,11 +1,22 @@
 #!/usr/bin/env ruby
 
+class Dog
+  attr_reader :name
+  def initialize(name)
+    @name = name
+  end
+end
+
 # Bring OptionParser in
 require "optparse"
 
 options = {}
 option_parser = OptionParser.new do |opts|
 
+  opts.accept(Dog) do |name|
```

Comment:
The argument passed on the command-line to the `-p` flag is what is passed as the block argument. On line 17 below, we can see that we are passing this to `Dog.new` which will initialize a new `Dog` object and store it in `options[:password]`. (the name of the key here for this example doesn't mean anything, I just used it to play around with getting this to work)